// actions/banner.js
import React, { Component } from 'react';
import { Image } from 'react-native'
import { createAction } from 'redux-actions';
import songService from 'services/song_service';
import Types from 'constants/types';

// Banner links
export function requestBannerLinks(){
  return createAction(Types.REQUEST_BANNER_LINKS)();
}

export function receiveBannerLinks(data){
  return createAction(Types.RECEIVE_BANNER_LINKS)(data);
}

export function bannerLinksError(message){
  return createAction(Types.BANNER_LINKS_ERROR)(message);
}

export function startLoadingFirstBanner(){
  return createAction(Types.START_LOADING_FIRST_BANNER)();
}

export function finishLoadingFirstBanner(){
  return createAction(Types.FINISH_LOADING_FIRST_BANNER)();
}

export function startLoadingSecondBanner(){
  return createAction(Types.START_LOADING_SECOND_BANNER)();
}

export function finishLoadingSecondBanner(){
  return createAction(Types.FINISH_LOADING_SECOND_BANNER)();
}

export function fetchBannerLinks(){
  return dispatch => {
    dispatch(requestBannerLinks());
    return songService.getBannerLinks((response, status) => {
      if (status === undefined || status != 200){
        dispatch(bannerLinksError(response.message))
        return;
      }
      dispatch(startLoadingFirstBanner())
      dispatch(startLoadingSecondBanner())
      Image.prefetch(response.first).then(() => dispatch(finishLoadingFirstBanner()))
      Image.prefetch(response.second).then(() => dispatch(finishLoadingSecondBanner()))
      dispatch(receiveBannerLinks(response));
    })
  }
}
