// actions/song_list.js
import { createAction } from 'redux-actions';
import songService from 'services/song_service';
import Types from 'constants/types';

// New songs
export function requestNewSongs(){
  return createAction(Types.REQUEST_NEW_SONGS)();
}

export function receiveNewSongs(data){
  return createAction(Types.RECEIVE_NEW_SONGS)(data);
}

export function requestAllNewSongs(){
  return createAction(Types.REQUEST_ALL_NEW_SONGS)();
}

export function receiveAllNewSongs(data){
  return createAction(Types.RECEIVE_ALL_NEW_SONGS)(data);
}

export function newSongError(message){
  return createAction(Types.NEW_SONGS_ERROR)(message);
}

export function allNewSongError(message){
  return createAction(Types.ALL_NEW_SONGS_ERROR)(message);
}

export function fetchNewSongs(){
  return dispatch => {
    dispatch(requestNewSongs());
    return songService.getNewSongs((response, status) => {
      if (status === undefined || status != 200){
        dispatch(newSongError(response.message))
        return;
      }
      dispatch(receiveNewSongs(response));
    })
  }
}

export function fetchAllNewSongs(){
  return dispatch => {
    dispatch(requestAllNewSongs());
    return songService.getAllNewSongs((response, status) => {
      if (status === undefined || status != 200){
        dispatch(allNewSongError(response.message))
        return;
      }
      dispatch(receiveAllNewSongs(response));
    })
  }
}

// Popular songs
export function requestPopularSongs(){
  return createAction(Types.REQUEST_POPULAR_SONGS)();
}

export function receivePopularSongs(data){
  return createAction(Types.RECEIVE_POPULAR_SONGS)(data);
}

export function requestAllPopularSongs(){
  return createAction(Types.REQUEST_ALL_POPULAR_SONGS)();
}

export function receiveAllPopularSongs(data){
  return createAction(Types.RECEIVE_ALL_POPULAR_SONGS)(data);
}

export function popularSongError(message){
  return createAction(Types.POPULAR_SONGS_ERROR)(message);
}

export function allPopularSongError(message){
  return createAction(Types.ALL_POPULAR_SONGS_ERROR)(message);
}

export function fetchPopularSongs(){
  return dispatch => {
    dispatch(requestPopularSongs());
    return songService.getPopularSongs((response, status) => {
      if (status === undefined || status != 200){
        dispatch(popularSongError(response.message))
        return;
      }
      dispatch(receivePopularSongs(response));
    })
  }
}

export function fetchAllPopularSongs(){
  return dispatch => {
    dispatch(requestAllPopularSongs());
    return songService.getAllPopularSongs((response, status) => {
      if (status === undefined || status != 200){
        dispatch(allPopularSongError(response.message))
        return;
      }
      dispatch(receiveAllPopularSongs(response));
    })
  }
}
