// reducers/banner.js
import React, { Component } from 'react';
import Types from 'constants/types';

const initialState = {
  links: {},
  loadingLinks: false,
  loadingLinkError: false,
  loadingLinkErrorMessage: '',
  loadingFirst: false,
  loadingSecond: false,
}

module.exports = (state=initialState, action) => {
  switch(action.type){
    case Types.REQUEST_BANNER_LINKS: {
      return Object.assign(
        {}, state, { loadingLinks: true }
      );
    }

    case Types.RECEIVE_BANNER_LINKS: {
      return Object.assign(
        {}, state, {
          loadingLinks: false,
          links: {
            first: action.payload.first,
            second: action.payload.second,
          },
        }
      );
    }

    case Types.BANNER_LINKS_ERROR: {
      return Object.assign(
        {}, state, { loadingLinks: false, loadingLinkError: true, loadingLinkErrorMessage: action.payload }
      );
    }

    case Types.START_LOADING_FIRST_BANNER: {
      return Object.assign(
        {}, state, { loadingFirst: true }
      );
    }

    case Types.FINISH_LOADING_FIRST_BANNER: {
      return Object.assign(
        {}, state, { loadingFirst: false }
      );
    }

    case Types.START_LOADING_SECOND_BANNER: {
      return Object.assign(
        {}, state, { loadingSecond: true }
      );
    }
    case Types.FINISH_LOADING_SECOND_BANNER: {
      return Object.assign(
        {}, state, { loadingSecond: false }
      );
    }

    default:
      return state;
  }
}
