// reducers/video.js
import Types from 'constants/types';
const initialState = {
  source: 'http://www.headerhandler.com/downloads/test.mp3',
  playing: false,
  repeat: false,
  currentTime: 0,
  duration: 0,
  stopped: false,
  component: undefined
}

module.exports = (state=initialState, action) => {
  switch(action.type){
    case Types.SET_VIDEO_SOURCE: {
      return Object.assign(
        {}, state, {
          source: action.payload,
          playing: true,
        }
      );
    }

    case Types.SET_VIDEO_CURRENT_TIME: {
      return Object.assign(
        {}, state, { currentTime: action.payload }
      );
    }

    case Types.SET_VIDEO_DURATION: {
      return Object.assign(
        {}, state, { duration: action.payload }
      );
    }

    case Types.SET_VIDEO_COMPONENT: {
      return Object.assign(
        {}, state, { component: action.payload }
      );
    }

    case Types.SET_VIDEO_REPEAT: {
      return Object.assign(
        {}, state, { repeat: action.payload }
      );
    }

    case Types.STOP_VIDEO: {
      return Object.assign(
        {}, state, {
          playing: false,
          stopped: true,
          currentTime: 0
        }
      );
    }

    case Types.PLAY_VIDEO: {
      return Object.assign(
        {}, state, {
          playing: true,
          stopped: false
        }
      );
    }

    case Types.PAUSE_VIDEO: {
      return Object.assign(
        {}, state, { playing: false }
      );
    }

    case Types.TOGGLE_VIDEO_PLAY: {
      const { playing } = state;
      return Object.assign(
        {}, state, { playing: !playing }
      );
    }

    default:
      return state;
  }
}
