// services/song_service.js
const apiUrl = "http://192.168.43.162:9192"

module.exports = {
  getNewSongs: function(callback){
    this.get('/api/songs/new_trimmed', callback);
  },

  getAllNewSongs: function(callback){
    this.get('/api/songs/new', callback);
  },

  getPopularSongs: function(callback){
    this.get('/api/songs/popular_trimmed', callback);
  },

  getAllPopularSongs: function(callback){
    this.get('/api/songs/popular', callback);
  },

  get: function(url, callback){
    fetch(apiUrl + url, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }
    })
    .then(response => {
      if (response.status == 200)
        response.json().then((response) => callback(response, 200))
      else
        response.json().then((response) => callback(response))
    })
    .catch(e => callback(e))
  },

  post: function(url, object, callback){
    fetch(apiUrl + url, {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(object)
    })
    .then(response => {
      if (response.status == 200)
        response.json().then((response) => callback(response, 200))
      else
        response.json().then((response) => callback(response))
    })
    .catch(e => callback(e))
  }
}
