// services/song_service.js
import Types from 'constants/types';
import urls from 'constants/urls';
import api from './api';

module.exports = {
  getNewSongs: function(requestType, callback){
    this.get(urls.new[requestType], callback);
  },

  getAllNewSongs: function(callback){
    this.get('/api/songs/new', callback);
  },

  getPopularSongs: function(requestType, callback){
    this.get(urls.popular[requestType], callback);
  },

  getAllPopularSongs: function(callback){
    this.get('/api/songs/popular', callback);
  },

  getBannerLinks: function(callback){
    this.get('/api/banners/new', callback)
  },

  get: function(url, callback){
    api.get(url, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }
    })
    .then(response => {
      if (response.status == 200)
        callback(response.data, 200)
      else
        callback(response.data)
    })
    .catch(e => callback(e))
  },

  post: function(url, object, callback){
    api.post(url, JSON.stringify(object), {
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      },
    })
    .then(response => {
      if (response.status == 200)
        response.json().then((response) => callback(response, 200))
      else
        response.json().then((response) => callback(response))
    })
    .catch(e => callback(e))
  }
}
